src/xrpld/ledger/detail/View.cpp

```diff
@@ -2692,6 +2705,163 @@ sharesToAssetsWithdraw(
     return assets;
 }
 
+TER
+rippleLockEscrowMPT(
+    ApplyView& view,
+    AccountID const& sender,
+    STAmount const& amount,
+    beast::Journal j)
+{
+    auto const mptIssue = amount.get<MPTIssue>();
+    auto const mptID = keylet::mptIssuance(mptIssue.getMptID());
+    auto sleIssuance = view.peek(mptID);
+    if (!sleIssuance)
+        return tecOBJECT_NOT_FOUND;
+
+    // 1. Decrease the MPT Holder MPTAmount
+    // 2. Increase the MPT Holder EscrowedAmount
+    if (amount.getIssuer() != sender)
+    {
```

Comment:
the issuer can never be the sender, code can be refactored
```suggestion
    if (amount.getIssuer() == sender)
        return tecINTERNAL;
        
    // 1. Decrease the MPT Holder MPTAmount
    // 2. Increase the MPT Holder EscrowedAmount
    {
```